pkg/tasks/local/ticket.go

```diff
@@ -18,6 +18,126 @@ import (
 	sdkTicket "github.com/conductorone/baton-sdk/pkg/types/ticket"
 )
 
+type localBulkCreateTicket struct {
+	o sync.Once
+
+	templatePath string
+}
+
+type bulkCreateTicketTemplate struct {
+	Tickets []ticketTemplate `json:"tickets"`
+}
+
+func (m *localBulkCreateTicket) loadTicketTemplate(ctx context.Context) (*bulkCreateTicketTemplate, error) {
+	tbytes, err := os.ReadFile(m.templatePath)
+	if err != nil {
+		return nil, err
+	}
+
+	template := &bulkCreateTicketTemplate{}
+	err = json.Unmarshal(tbytes, template)
+	if err != nil {
+		return nil, err
+	}
+
+	return template, nil
+}
+
+func (m *localBulkCreateTicket) GetTempDir() string {
+	return ""
+}
+
+func (m *localBulkCreateTicket) ShouldDebug() bool {
+	return false
+}
+
+func (m *localBulkCreateTicket) Next(ctx context.Context) (*v1.Task, time.Duration, error) {
+	var task *v1.Task
+	m.o.Do(func() {
+		task = &v1.Task{
+			TaskType: &v1.Task_BulkCreateTickets{
+				BulkCreateTickets: &v1.Task_BulkCreateTicketsTask{},
+			},
+		}
+	})
+	return task, 0, nil
+}
+
+func (m *localBulkCreateTicket) Process(ctx context.Context, task *v1.Task, cc types.ConnectorClient) error {
+	l := ctxzap.Extract(ctx)
+
+	templates, err := m.loadTicketTemplate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticketReqs := make([]*v2.TicketsServiceCreateTicketRequest, 0)
+	for _, template := range templates.Tickets {
+		schema, err := cc.GetTicketSchema(ctx, &v2.TicketsServiceGetTicketSchemaRequest{
+			Id: template.SchemaID,
+		})
+		if err != nil {
+			return err
+		}
+
+		ticketRequestBody := &v2.TicketRequest{
+			DisplayName: template.DisplayName,
+			Description: template.Description,
+			Labels:      template.Labels,
+		}
+
+		if template.StatusId != "" {
+			ticketRequestBody.Status = &v2.TicketStatus{
+				Id: template.StatusId,
+			}
+		}
+
+		if template.RequestedForId != "" {
+			rt := resource.NewResourceType("User", []v2.ResourceType_Trait{v2.ResourceType_TRAIT_USER})
+			requestedUser, err := resource.NewUserResource(template.RequestedForId, rt, template.RequestedForId, []resource.UserTraitOption{})
+			if err != nil {
+				return err
+			}
+			ticketRequestBody.RequestedFor = requestedUser
+		}
+
+		cfs := make(map[string]*v2.TicketCustomField)
+		for k, v := range template.CustomFields {
+			newCfs, err := sdkTicket.CustomFieldForSchemaField(k, schema.Schema, v)
+			if err != nil {
+				return err
+			}
+			cfs[k] = newCfs
+		}
+		ticketRequestBody.CustomFields = cfs
+
+		ticketReqs = append(ticketReqs, &v2.TicketsServiceCreateTicketRequest{
+			Request: ticketRequestBody,
+			Schema:  schema.GetSchema(),
+		})
+	}
+
+	bulkTicketReq := &v2.TicketsServiceBulkCreateTicketRequest{
+		TicketRequests: ticketReqs,
+	}
+
+	resp, err := cc.BulkCreateTickets(ctx, bulkTicketReq)
+	if err != nil {
+		return err
```

Comment:
Hrm, do we want to attempt all the tickets and report a list of errors or one ticket fails they all fail? This would cause all ticketing to fail for zscaler because one config went out of date, and they have many. I don't think we should kill the full batch. All tickets would get stuck this way.

Reply:
This does not indicate that we do fail for individual tickets. It is making a BulkCreateTickets call here.

Reply:
And this is local